json_object.h

```diff
@@ -718,6 +718,22 @@ JSON_EXPORT int32_t json_object_get_int(const struct json_object *obj);
  */
 JSON_EXPORT int json_object_set_int(struct json_object *obj,int new_value);
 
+/** Add the int value to the value of a json object
+ *
+ * The type of obj is checked to be a json type int and 0 is returned
+ * if it is not without any further actions. If the type of obj is
+ * json_type_int the int value is added to the object value.
+ * If the addition would result in a overflow, the object value
+ * is set to INT32_MAX.
+ * If the addition would result in a underflow, the object value
+ * is set to INT32_MIN.
+ *
+ * @param obj the json_object instance
+ * @param val the value to add
+ * @returns 1 if the addition succeded, 0 otherwise
+ */
+JSON_EXPORT int json_object_add_int(struct json_object *obj, int val);
```

Comment:
maybe more of a preference [from me] here, but, keeping the name `json_object_inc_int()` [like mentioned in issue #338 ] or similar, could be a good idea, because it makes it clearer that it's incrementing an int value;

in the context of `libjson-c` a name that like `json_object_add_xxx` reminds me of the `json_object_object_add()` or `json_object_array_add()` functions where a JSON object is added to a another JSON object

i'd also rename the `int val` param to `int inc`
and you can also create a convenience macro called `json_object_dec_int()` where
```
#define json_object_dec_int(x,y)    json_object_inc_int(x, -(y))
```

but these are all just preferences/thoughts :)